### DISCUSSION TITLE
Ultralytics installs torch 2.0 even though I have torch 1.13

### DISCUSSION THREAD

**Discussion by suchith-sixsense** [ultralytics/ultralytics#3002] [General]
As per the screenshot below, in the first line it says that `torch>1.7.0` requirement is satisfied. But at the end if you see, it collects `torch 2.0` and installs it while uninstalling it `torch 1.13`.

Can I know why this is happening and can I do something to avoid this ?

<img width="1439" alt="Screenshot 2023-06-04 at 11 54 59 PM" src="https://github.com/ultralytics/ultralytics/assets/110653781/1081f630-fba6-4e1e-8857-1e556708522c">

Thanks.

**Comment by glenn-jocher** [comment]
@suchith-sixsense hello! It seems like the installation process is upgrading your PyTorch version to meet the requirements of another dependency that requires `torch 2.0`. This can happen if there's a dependency in the YOLOv8 requirements that needs a newer version of PyTorch.

To avoid this automatic upgrade, you could manually manage your environment. One approach is to create a separate virtual environment for YOLOv8 and install the specific versions of the libraries you need. This way, you can ensure that the installation of YOLOv8 does not interfere with your existing setup.

Remember to always check the `requirements.txt` file for the specific versions of the dependencies that YOLOv8 needs. If you need to maintain a specific version of PyTorch for your projects, you might need to adjust other dependencies accordingly to ensure compatibility.

If you need further guidance on managing dependencies or setting up your environment, you can refer to our documentation at https://docs.ultralytics.com. 😊
